kong/plugins/acme/storage/config_adapters/redis.lua

```diff
@@ -3,13 +3,13 @@ local function redis_config_adapter(conf)
         host = conf.host,
         port = conf.port,
         database = conf.database,
-        auth = conf.password or conf.auth, -- allow conf.auth until 4.0 version
+        auth = conf.password,
         ssl = conf.ssl,
         ssl_verify = conf.ssl_verify,
-        ssl_server_name = conf.server_name or conf.ssl_server_name, -- allow conf.ssl_server_name until 4.0 version
+        ssl_server_name = conf.server_name,
 
-        namespace = conf.extra_options.namespace or conf.namespace, -- allow conf.namespace until 4.0 version
-        scan_count = conf.extra_options.scan_count or conf.scan_count, -- allow conf.scan_count until 4.0 version
+        namespace = conf.extra_options.namespace,
+        scan_count = conf.extra_options.scan_count,
```

Comment:
I agree on the path towards unification of config fields, but I'm wondering whether the new names read better than the old ones. For example, changing `ssl-server_name` to `server_name` seems making the name more ambigious, and an additional `extra_options` level seems unnessary.
As we are going to bring this breaking change in 4.0 anyway, does it make sense to change the `kong/enterprise_edition/init.lua` side instead? How do you think?

Reply:
On the other side, (I just caught up the design doc), as we are planning to add
redis configuration to a first class entity, should we just drop all the redis field from
plugin config and allows only `config.redis.id = <the redis entity ID>` form instead
of bringing this transitional change?

Reply:
The transitional step is being done to simplify the move from a plugin-individual redis configuration to a unified-global redis configuration.

In this step, we're moving from the `ssl-server_name` that is a non-prefixed redis specific field to `redis.server_name` for example.
The next step (implementing redis partials), will replace `redis` block with a reference, while maintaining the strucutre.

Regarding the `extra_options` - Some plugins have unique configuration options that are not unifiable in a global schema, so there needs to be a mechanism to include custom options in there.